contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,507 @@
+     */
+    function getSchedule(address _beneficiary, uint256 _index) external view returns(uint256, uint256, uint256, uint256, State) {
+        require(_beneficiary != address(0), "Invalid address");
+        require(_index < schedules[_beneficiary].length, "Schedule not found");
+        Schedule storage schedule = schedules[_beneficiary][_index];
+        return (
+            schedule.numberOfTokens,
+            schedule.duration,
+            schedule.frequency,
+            schedule.startTime,
+            _getScheduleState(_beneficiary, _index)
+        );
+    }
+
+    function _getScheduleState(address _beneficiary, uint256 _index) internal view returns(State) {
+        Schedule memory schedule = schedules[_beneficiary][_index];
+        if (now < schedule.startTime) {
+            return State.CREATED;
+        } else if (now > schedule.startTime && now < schedule.startTime.add(schedule.duration)) {
+            return State.STARTED;
+        } else {
+            return State.COMPLETED;
+        }
+    }
+
+    /**
+     * @notice Returns count of beneficiary's schedules
+     * @param _beneficiary beneficiary's address
+     * @return count of beneficiary's schedules
+     */
+    function getScheduleCount(address _beneficiary) external view returns(uint256) {
+        require(_beneficiary != address(0), "Invalid address");
+        return schedules[_beneficiary].length;
+    }
+
+    /**
+     * @notice Returns available tokens for beneficiary
+     * @param _beneficiary beneficiary's address
+     * @return available tokens for beneficiary
+     */
+    function getAvailableTokens(address _beneficiary) external view returns(uint256) {
+        return _getAvailableTokens(_beneficiary);
+    }
+
+    function _getAvailableTokens(address _beneficiary) internal view returns(uint256) {
+        require(_beneficiary != address(0));
+        uint256 availableTokens;
+        for (uint256 i = 0; i < schedules[_beneficiary].length; i++) {
+            availableTokens = availableTokens.add(schedules[_beneficiary][i].availableTokens);
+        }
+        return availableTokens;
+    }
+
+    /**
+     * @notice Used to bulk send available tokens for each of beneficiaries
+     * @param _beneficiaries array of beneficiary's addresses
+     */
+    function pushAvailableTokensMulti(address[] _beneficiaries) external withPerm(ADMIN) {
+        require(_beneficiaries.length > 1, "Array size should be greater than one");
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            pushAvailableTokens(_beneficiaries[i]);
+        }
+    }
+
+    /**
+     * @notice Used to bulk add vesting schedules for each of beneficiaries
+     * @param _beneficiaries array of beneficiary's addresses
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     * @param _startTime vesting start time
+     */
+    function addScheduleMulti(
```

Comment:
What about startTime?
I think we can change this function to use arrays instead of single values.
Adding the same schedules for different beneficiaries is possible using templates.
We need to optimize contact size, adding new function produces "out of gas" error during contract creation.